### PR TITLE
MWPFE-17-header component, button component, images folder

### PR DESCRIPTION
**Pull request by medeinedapkev**
Created header component, button component for header button "Prisijungti" and also created folder images for companies logo images. Installed headlessui/react, heroicons/react, lucide-react for icons and components.

### CODE REVIEW

File: src/components/ui/Header.jsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+function Header() {
+  const [mobileMenuOpen, setMobileMenuOpen] = useState(false);
+
+  return (
+    <header className="bg-white">
+      <nav className="mx-auto flex max-w-7xl items-center justify-between p-6 lg:px-8" aria-label="Global">
+        <div className="flex lg:flex-1">
+          <Link href="/" className="-m-1.5 p-1.5">
+            <span className="sr-only">Your Company</span>
+            <img className="h-8 w-auto" src={Logo.src} alt="logo" />
```
Comment (lauramatiukievic) [inline src/components/ui/Header.jsx:20]: reiketu naudoti  Image taga is nexto https://nextjs.org/docs/app/building-your-application/optimizing/images

**Review by lauramatiukievic — COMMENTED** [review]
visa kita man atrodo goood

File: src/components/ui/Header.jsx (hunk 1 above)
Comment (medeinedapkev) [reply src/components/ui/Header.jsx]: Fixed, new commit

**Review by dkisonas — CHANGES_REQUESTED** [review]

File: src/app/HeaderContext.jsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import { createContext, useContext, useState } from 'react';
+
+const MobileMenuContext = createContext();
+
+export const useMobileMenuContext = () => {
+  return useContext(MobileMenuContext);
+};
+
+export const MobileMenuProvider = ({ children }) => {
```
Comment (dkisonas) [inline src/app/HeaderContext.jsx:11]: File called HeaderContext, should be MobileMenuProvider

Also should be moved to components folder

File: .env.example
Comment (dkisonas) [inline .env.example:1]: don't delete this file

File: src/components/ui/Header.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        </div>
+        <div className="hidden lg:flex lg:flex-1 lg:justify-end">
+          <Link href="#" className="text-sm font-semibold leading-6 text-gray-900">
+            <Button>{headerLogIn.name}</Button>
+          </Link>
+        </div>
+      </nav>
+
+      {headerLink.map(data => (
+        <HeaderDialog data={data} company={headerLogo} logIn={headerLogIn} />
```
Comment (dkisonas) [inline src/components/ui/Header.jsx:42]: pass data only

File: src/components/ui/Header.jsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        <div className="flex lg:flex-1">
+          <Link href="/" className="-m-1.5 p-1.5">
+            <span className="sr-only">{headerLogo.name}</span>
+            <Image src={`${headerLogo.logo.url}`} alt="logo" width="40" height="40" />
+          </Link>
+        </div>
+
+        <HeaderMobileButton />
+        
+        {headerLink.map(data => (
```
Comment (dkisonas) [inline src/components/ui/Header.jsx:26]: headerLinks (with S at the end, beause many links)

File: src/components/ui/HeaderDialog.jsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+          onClick={closeMobileMenu}
+        >
+          <span className="sr-only">Close menu</span>
+          <XMarkIcon className="h-6 w-6" aria-hidden="true" />
+        </button>
+      </div>
+      <div className="mt-6 flow-root">
+        <div className="-my-6 divide-y divide-gray-500/10">
+          <div className="space-y-2 py-6">
+            {links.links.map(link => (
```
Comment (dkisonas) [inline src/components/ui/HeaderDialog.jsx:38]: links.links? something not right with naming here

File: src/components/ui/HeaderPopover.jsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
+'use client'
+
+import { Popover } from '@headlessui/react';
+import Link from 'next/link';
+
+const HeaderPopover = ({ data: links }) => {
+
+  return (
+    <Popover.Group className="hidden lg:flex lg:gap-x-12">
+      {links.links.map(link => (
```
Comment (dkisonas) [inline src/components/ui/HeaderPopover.jsx:10]: links.links, change naming

File: src/components/ui/Header.jsx [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+import { Button } from "@/components/ui/Button";
+
+import Image from 'next/image';
+import Link from 'next/link';
+import HeaderPopover from './HeaderPopover';
+import HeaderDialog from './HeaderDialog';
+import HeaderMobileButton from './HeaderMobileButton';
+
+function Header({ data }) {
+  console.log(data)
```
Comment (dkisonas) [inline src/components/ui/Header.jsx:11]: remove console.log

**Review by dkisonas — APPROVED** [review]
approved with few small suggestions

File: src/app/HeaderContext.jsx (hunk 2 above)
Comment (medeinedapkev) [reply src/app/HeaderContext.jsx]: fixed

File: src/components/ui/Header.jsx (hunk 7 above)
Comment (medeinedapkev) [reply src/components/ui/Header.jsx]: removed

File: src/components/ui/HeaderPopover.jsx (hunk 6 above)
Comment (medeinedapkev) [reply src/components/ui/HeaderPopover.jsx]: fixed

File: src/components/ui/HeaderDialog.jsx (hunk 5 above)
Comment (medeinedapkev) [reply src/components/ui/HeaderDialog.jsx]: fixed

File: src/components/ui/Header.jsx (hunk 4 above)
Comment (medeinedapkev) [reply src/components/ui/Header.jsx]: fixed

File: src/components/ui/Header.jsx (hunk 3 above)
Comment (medeinedapkev) [reply src/components/ui/Header.jsx]: fixed

File: .env.example
Comment (medeinedapkev) [reply .env.example:1]: fixed

**Review by dkisonas — APPROVED** [review]

**Pull request merged**